datalad_registry/blueprints/api/dataset_urls.py

```diff
@@ -0,0 +1,278 @@
+    """
+    if isinstance(url, Path) and not url.is_absolute():
+        raise ValueError("Path URLs must be absolute")
+    return url
+
+
+class _PathParams(BaseModel):
+    """
+    Pydantic model for representing the path parameters for API endpoints related to
+    dataset URLs.
+    """
+
+    id: int = Field(..., description="The ID of the dataset URL")
+
+
+class _QueryParams(BaseModel):
+    """
+    Pydantic model for representing the query parameters to query
+    the dataset_urls endpoint
+    """
+
+    url: Optional[Union[FileUrl, AnyUrl, Path]] = Field(None, description="The URL")
+
+    ds_id: Optional[UUID] = Field(None, description="The ID, a UUID, of the dataset")
+
+    min_annex_key_count: Optional[int] = Field(
+        None, description="The minimum number of annex keys "
+    )
+    max_annex_key_count: Optional[int] = Field(
+        None, description="The maximum number of annex keys "
+    )
+
+    min_annexed_files_in_wt_count: Optional[int] = Field(
+        None, description="The minimum number of annexed files in the working tree"
+    )
+    max_annexed_files_in_wt_count: Optional[int] = Field(
+        None, description="The maximum number of annexed files in the working tree"
+    )
+
+    min_annexed_files_in_wt_size: Optional[int] = Field(
+        None,
+        description="The minimum size of annexed files in the working tree in bytes",
+    )
+    max_annexed_files_in_wt_size: Optional[int] = Field(
+        None,
+        description="The maximum size of annexed files in the working tree in bytes",
+    )
+
+    earliest_last_update: Optional[datetime] = Field(
+        None,
+        description="The earliest last update time",
+    )
+    latest_last_update: Optional[datetime] = Field(
+        None,
+        description="The latest last update time",
+    )
+
+    min_git_objects_kb: Optional[int] = Field(
+        None, description="The minimum size of the `.git/objects` in KiB"
+    )
+    max_git_objects_kb: Optional[int] = Field(
+        None, description="The maximum size of the `.git/objects` in KiB"
+    )
+
+    processed: Optional[bool] = Field(
+        None,
+        description="Whether an initial processing has been completed "
+        "on the dataset URL",
+    )
+
+    # Validator
+    _path_url_must_be_absolute = validator("url", allow_reuse=True)(
+        path_url_must_be_absolute
+    )
+
+
+class DatasetURLSubmitModel(BaseModel):
+    """
+    Model for representing the database model URL for submission communication
+    """
+
+    url: Union[FileUrl, AnyUrl, Path] = Field(..., description="The URL")
+
+    # Validator
+    _path_url_must_be_absolute = validator("url", allow_reuse=True)(
+        path_url_must_be_absolute
+    )
+
+
+class DatasetURLRespModel(DatasetURLSubmitModel):
+    """
+    Model for representing the database model URL in response communication
+    """
+
+    id: int = Field(..., alias="id", description="The ID of the dataset URL")
```

Comment:
Using `...` instead of `None` makes the field as required. For more information, please checkout https://docs.pydantic.dev/usage/models/#required-fields and https://docs.pydantic.dev/usage/schema/#field-customization, and try these examples.

```python
from typing import Optional
from pydantic import BaseModel, Field


class WithRequiredField(BaseModel):
    """
    `foo = WithRequiredField()` raises a ValidationError
    """
    a: str = Field(...)

class WithNoRequiredField(BaseModel):
    """
    `foo = WithNoRequiredField()` is allowed and sets `a` to None
    """
    a: str = Field(None)

class WithNoDefault(BaseModel):
    """
    `foo = WithNoDefault()` raises a ValidationError
    """
    a: str = Field()

class WithNoDefaultButMarkedOptional(BaseModel):
    """
    `foo = WithNoDefaultButMarkedOptional()` is allowed and sets `a` to None
    """
    a: Optional[str] = Field()
    
class MarkedOptionalButMarkedRequired(BaseModel):
    """
    `foo = MarkedOptionalButMarkedRequired()` raises a ValidationError
    """
    a: Optional[str] = Field(...)
```

I think this notation exists mainly for the case demonstrated in the `MarkedOptionalButMarkedRequired` example.